src/Day1.hs

```diff
@@ -3,18 +3,17 @@ module Day1 (run) where
 import Utils (parseIntsFromStrings, readInputFile)
 
 part1 :: [Int] -> Int
-part1 x = part1' 0 x
+part1 x = fst $ foldl part1' (0, -1) x
 
-part1' :: Int -> [Int] -> Int
-part1' a [] = a
-part1' a (l: ls) = if length ls > 0 then part1' (a + (if l >= (head ls) then 0 else 1)) ls else a
+-- increases, prev_depth
+part1' :: (Int, Int) -> Int -> (Int, Int) 
+part1' (acc, -1) newDepth = (acc, newDepth)
```

Comment:
In general, you want to avoid using "special values" as sentinels. This tends to be a code smell. What if one day `-1` actually _does_ become a valid depth measurement?

Instead, you can use something like `Maybe Int` to make this code a lot safer for relatively low cost by explicitly handling the "I don't have a depth measurement yet" case.

(In this case however, just take a quick glance at your input data - is `depth` ever lower than `0`? If not, can we simplify our code here?)

Reply:
depth is never lower than 0, the only reason for including that was to handle the first case. i knew it was slightly sloppy but wanted to test logic, refactoring it now

Reply:
It looks like you've decided to go with the `Maybe` case, which is great! That's the maximally safe way to solve this issue.

The clever simplification given that no `depth` is ever lower than `0` is to just set the initial depth at `0`. Note that this only works for data sets where this invariant (`depth >= 0`) holds, and will break silently when that invariant is violated!

```hs
-- I've also eta-reduced into point-free notation here.
part1 = fst . foldl f (0, 0)
  where
    f (increases, prevDepth) newDepth = (increases + if newDepth > prevDepth then 1 else 0, newDepth)
```

This is a simpler but less safe solution than using `Maybe`, but IMO works fine for things like Advent where you have very strong guarantees about your input shape.